APE12.rst

```diff
@@ -0,0 +1,209 @@
+Using Cookiecutter for the package-template
+-------------------------------------------
+
+author: Stuart Mumford, Thomas Robitaille
+
+date-created: 2016 November 17
+
+date-last-revised: 2017 January 24
+
+type: Standard Track
+
+status: Discussion
+
+
+Abstract
+--------
+
+This APE proposes migrating the `package-template
+<https://github.com/astropy/package-template/>`_ to use the `Cookiecutter
+<http://cookiecutter.readthedocs.io/>`_ project to provide templating. This
+would enable users to be able to customise the package template for their
+package from the command line, in a much simpler manner than the current manual
+process. As well as this, having a templating system makes it easier for other
+projects to change the branding or default settings of the package template.
+
+This APE proposes major changes to the current layout of the package-template
+repository, potentially breaking existing user workflows, hence the APE.
+
+
+Detailed description
+--------------------
+
+Cookiecutter is a Python project that uses the Jinja2 templating engine to
+define project templates for both Python and non-Python projects. When run it
+presents the user with a simple set of questions to populate predefined
+variables which are then used to render the Jinja2 templates and create the
+project. This workflow is a massive improvement on the manual workflow currently
+recommended to affiliated package authors in the
+`Astropy documentation <http://docs.astropy.org/en/latest/development/affiliated-packages.html>`_.
+
+The second advantage to using a templating system like Jinja2 is that it makes
+it possible for other projects (*e.g.* SunPy) to maintain a fork of the package
+template with defaults that specific to that project. Without having to make
+extensive changes to the actual package template, thereby making it easy to pull
+in changes from the Astropy template without lots of complex merging.
+
+There are also secondary advantages to using a Cookiecutter project, such as
+being able to include code i.e. tests, in the `package-template
+<https://github.com/astropy/package-template/>`_ repository while not including
+it in the code that the person setting up a new project receives. This is
+because only the templated directory in the repository gets processed by
+Cookiecutter.
+
+An example of using cookiecutter with the package template is below. This
+example uses the `Cadair/pacakage-template <https://github.com/Cadair/package-template/tree/cookiecutter>`_
+``cookiecutter`` branch, if this work was to be merged the command would be
+``cookiecutter gh:astropy/package-template`` as the ``-c`` option specifies a
+non-master branch.
```

Comment:
@Cadair - very minor before I approve and merge, we should keep the -c cookiecutter since we're going to merge this into a cookiecutter branch. So can you update it here and in the first command of the code block below?